### ISSUE TITLE
[vmagent] behaves differently on relabel configuration order than prometheus.

### ISSUE THREAD

**Issue by zbindenren** [VictoriaMetrics/VictoriaMetrics#812]
**Describe the bug**
vmagent behaves differently on relabel configuration order than prometheus.

**To Reproduce**
The following scrape configuration:
```yaml
scrape_configs:
  - job_name: node-exporter
    honor_timestamps: true
    scrape_interval: 20s
    scrape_timeout: 10s
    metrics_path: /metrics
    scheme: http
    kubernetes_sd_configs:
    - role: pod
    relabel_configs:
    - separator: ;
      regex: __meta_kubernetes_pod_label_(.+)
      replacement: $1
      action: labelmap
    - source_labels: [mountpoint]
      action: drop
      regex: (.+)
    - source_labels: [__address__]
      separator: ;
      regex: ([^:]+)(?::\d+)?
      target_label: __address__
      replacement: $1:9100
      action: replace
    - source_labels: [__meta_kubernetes_namespace, __meta_kubernetes_pod_label_k8s_app]
      separator: ;
      regex: kube-metrics;node-exporter
      replacement: $1
      action: keep
    - source_labels: [__meta_kubernetes_namespace]
      separator: ;
      regex: (.*)
      target_label: namespace
      replacement: $1
      action: replace
    - source_labels: [__meta_kubernetes_pod_name]
      separator: ;
      regex: (.*)
      target_label: pod
      replacement: $1
      action: replace
    - source_labels: [__meta_kubernetes_pod_node_name]
      separator: ;
      regex: (.*)
      target_label: instance
      replacement: $1
      action: replace
```

works with prometheus, but does not work with vmagent -> there are no targets.

If I move up the block with the `keep` action:
```yaml
    - source_labels: [__meta_kubernetes_namespace, __meta_kubernetes_pod_label_k8s_app]
      separator: ;
      regex: kube-metrics;node-exporter
      replacement: $1
      action: keep
```
like:

```yaml
scrape_configs:
  - job_name: node-exporter
    honor_timestamps: true
    scrape_interval: 20s
    scrape_timeout: 10s
    metrics_path: /metrics
    scheme: http
    kubernetes_sd_configs:
    - role: pod
    relabel_configs:
    - source_labels: [__meta_kubernetes_namespace, __meta_kubernetes_pod_label_k8s_app]
      separator: ;
      regex: kube-metrics;node-exporter
      replacement: $1
      action: keep
    - separator: ;
      regex: __meta_kubernetes_pod_label_(.+)
      replacement: $1
      action: labelmap
    - source_labels: [mountpoint]
      action: drop
      regex: (.+)
    - source_labels: [__address__]
      separator: ;
      regex: ([^:]+)(?::\d+)?
      target_label: __address__
      replacement: $1:9100
      action: replace
    - source_labels: [__meta_kubernetes_namespace]
      separator: ;
      regex: (.*)
      target_label: namespace
      replacement: $1
      action: replace
    - source_labels: [__meta_kubernetes_pod_name]
      separator: ;
      regex: (.*)
      target_label: pod
      replacement: $1
      action: replace
    - source_labels: [__meta_kubernetes_pod_node_name]
      separator: ;
      regex: (.*)
      target_label: instance
      replacement: $1
      action: replace
```

it works also with vmagent.

**Expected behavior**
The relabel order behavior should be the same for vmagent and prometheus.

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
vmagent-20200929-220206-tags-v1.42.0-cluster-0-g081c5d2c
```

**Comment by f41gh7** [comment]
Seems like `labelmap` modifies `__meta_kubernetes_pod_label_` instead of adding new one. That's why `keep` action doesn't work.

**Comment by valyala** [comment]
The issue should be fixed in the following commits:

* cluster version - 7505e4f390b303fe8267c27d8aee771ac8e1d06a
* single-node version - f42194d817f74578b3ed3ca6a593b3f8ba9af3e4

**Comment by valyala** [comment]
@zbindenren , could you build vmagent from the commit https://github.com/VictoriaMetrics/VictoriaMetrics/commit/f42194d817f74578b3ed3ca6a593b3f8ba9af3e4 and verify whether your  relabeling case works as expected there? See [build instructions](https://victoriametrics.github.io/vmagent.html#how-to-build-from-sources).

**Comment by zbindenren** [comment]
@valyala I can confirm that it works with `master` (`vmagent-20201005-142442-heads-master-0-gcbe3cf68`).

**Comment by valyala** [comment]
Thanks for the confirmation! Then the bugfix will be included in the next release.

**Comment by valyala** [comment]
The bugfix has been included in [release v1.43.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the bug as fixed.
